fix: fallback if os.Rename() fails

leonbotros opened the pull request:
Should fix #107

leonbotros commented on internal/common/common.go:
```diff
@@ -7,14 +7,14 @@ import (
 	"fmt"
+	"github.com/go-errors/errors"
+	"github.com/privacybydesign/gabi/big"
+	"github.com/sirupsen/logrus"
```
Hi David, me and Sietse discussed to redo this PR to handle moving scheme directories the same way as files, i.e., in the current folder. This means the file cannot is very likely on the same filesystem and we can use `os.rename`. It also means we disregard the `conf.TempPath` and we end up with an atomic operation (no half scheme left behind in case of errors). Does someone know if `os` offers a more robust way to do this in general (i.e., rollback if something has gone wrong)?

As for the imports, I forgot to run my formatter :)

davidv1992 replied:
Ah ok, in that case i would suggest we mark this PR as a draft to indicate it shouldn't be merged

leonbotros commented:
I've updated this PR to use the same directory that old scheme is already using in `updateSchemeDir`.
This means no directories can be "moved" (i.e., renamed) across filesystems.
The function `tempSchemeCopy` does not need this because the `os.Rename` it performs is fully in the `/tmp` folder.
`conf.TempPath` is still used in this function so I think it should stay.

leonbotros commented:
I've given this some more thought. It turns out `tempSchemeCopy()` does in fact copy from `conf.TempPath` to the configuration directory. When we fully remove `conf.TempPath`, a new scheme is downloaded within this (configuration) directory and renamed after it is completed. The configuration `conf.TempPath` was primarily used for irmamobile where it was effectively creating a temporary folder within the `/data/data/<package>` instead of the system's  `/tmp` folder. Therefore, I think the configuration setting can be removed.

leonbotros commented:
@davidv1992 , what do you think about going this route, i.e. removing `conf.TempPath`? In my opinion this setting should've been used with `getExternalCacheDir()` on Android (note that this does not require extra permissions from API 19 and up).

sietseringers reviewed (changes requested):
There are compilation errors in `internal/sessiontest/main_test.go` and `irmaclient/irmaclient_test.go:45:20`.
